core/src/main/java/org/mule/api/MessagingException.java

```diff
@@ -390,17 +390,34 @@ protected void extractMuleMessage(MuleEvent event)
     private void writeObject(ObjectOutputStream out) throws Exception
     {
         out.defaultWriteObject();
-        if (this.failingMessageProcessor instanceof Serializable)
-        {
+        if (isSerializable(this.failingMessageProcessor)) {
             out.writeBoolean(true);
             out.writeObject(this.failingMessageProcessor);
-        }
-        else
-        {
+        } else {
             out.writeBoolean(false);
         }
     }
 
+    private boolean isSerializable(Object object) {
+        if (!(object instanceof Serializable)) {
+            return false;
+        }
+
+        Serializable serializableObject = (Serializable) object;
+        try {
+            Field[] fields = serializableObject.getClass().getDeclaredFields();
+            for (Field field : fields) {
+                field.setAccessible(true);
+                if (!(field.get(serializableObject) instanceof Serializable)) {
```

Comment:
Shouldn't this be a recursive call?

Reply:
Also, add tests for the recursive case (serializable with a serializable field that has a non-serializable field)